### DISCUSSION TITLE
browsers maximum limit for scrollable height forces to implement the scroll mechanism itself

### DISCUSSION THREAD

**Discussion by tschroeter** [petyosi/react-virtuoso#404] [Ideas]
There is no way to overcome the maximum pixel height (or width) to an element using browsers native scrolling capabilities.
Chrome for example doesn't accept a height greater than 33.554.400px.

For example, if we assume an average row height of 200px, the maximum number of items is just 167.772.

At the latest at this scale all Browser Engines will fail! Realistically, the engine already lacks significantly before.

For chat applications, these orders of magnitude could already be reached.

To overcome this issue the only solution is to implement the scrolling mechanisms itself as a native ui toolkit would do

**Comment by petyosi** [comment]
I have had similar discussions in the past. Reinventing the scroll mechanism is not something that is in the scope of this project - nor something which I personally want to do, ever. In fact, I believe that this would be downright harmful to the end-user. 

I would rather not spend my time arguing with people about it. Feel free to fork or start a new project around the concept.
